### ISSUE TITLE
[Question]: Babelfish update problem. how to update only Babelfish and keep postgres data.

### ISSUE THREAD

**Issue by weiya1990** [babelfish-for-postgresql/babelfish_extensions#91]
### Contact Details

1830624909@qq.com

### What's the question?

When the extension is updated. Use \dx to check the babelfish version is still old, if I drop extension  will  dependent objects. 

![image](https://user-images.githubusercontent.com/38902618/158748748-c0a77d60-2f06-4a33-8bd4-49c50bea74c5.png)

### Relevant log output or information
![image](https://user-images.githubusercontent.com/38902618/158752923-5a93028d-bbfc-4704-b72e-38e9f49c7a9a.png)

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by wieck** [comment]
Looks like one or more of your tables use a construct of `'text' + 'more_text'` in the DEFAULT clause. The upgrade .sql script wants to DROP and CREATE that + operator, which of course creates a dependency conflict.

Check file *babelfishpg_common/sql/upgrades/babelfishpg_common--1.1.0--1.2.0.sql* line 874.

I am not sure if the upgrade .sql could be improved to use different commands. What you could do is to simplify the DEFAULT clause to not use the text+text operator.

**Comment by sangilsong** [comment]
We found the issue that string concat operator '+' had returned wrong datatype. We fixed the problem by making the concat operator returning the correct datatype.
But, Postgres doesn't allow to change the return-datatype of operator in place (no CREATE OR REPLACE OPERATOR). It's because changing return-type of operator (or function) can affect all the depending objects, which is inappropriate to be executed in CREATE OR REPLACE OPERATOR. So upgrade script drops operator first and create new operator with the correct datatype. If depending objects exist, this will fail.
I understand that this is annoying to existing user, but returning a wrong datatype is correctness issue, so it is inevitable.
If upgrade fails by this reason, customer should dump the existing data, create latest extension from scratch and restore the data. Sorry for the inconvenience.
